### PR TITLE
Fix length and last packet issue in router

### PR DESCRIPTION
**Pull request by evoggy**
This avoids the asserts in the sending when splitting packets as well as properly sets the last packet flag.

### CODE REVIEW

File: main/router.c [hunk 1]
```diff
@@ -78,6 +78,8 @@ static void splitAndSend(const CPXRoutablePacket_t* rxp, RouteContext_t* context
     if (toSend > mtu) {
       toSend = mtu;
       lastPacket = false;
+    } else {
+      lastPacket = true;
```
Comment (krichardsson) [inline main/router.c:82]: I don't think lastPacket should be set to true here, instead it should be set based on the incoming packet (as it used to be). The reason is that the incoming packet may have been split in an earlier router and setting it explicitly to true here may make it hard for the receiving application to understand how to assemble data. Only passing the incoming value on will make sure the end receiver only gets one lastPacket==true per transmitted packet (from the original transmitter)

**Review by krichardsson — CHANGES_REQUESTED** [review]
Good that you found the packet sizes, we should really try to get unit tests in!
See comment for lastPacket.

I assume the discovery stuff was added unintentionally?

**Comment by evoggy** [comment]
New version where lastPacket is set on packages moving from the ESP APP into the route instead.

File: main/router.c (hunk 1 above)
Comment (evoggy) [reply main/router.c]: Yes, after discussing this there's a new commit for this.

Comment (evoggy) [reply main/router.c]: And yes, the discovery stuff was unintentional. I branched off from the wrong branch.

**Pull request merged**